modules/api/src/main/java/org/eclipse/xpanse/api/controllers/CspServiceTemplateApi.java

```diff
@@ -117,42 +120,62 @@ public List<ServiceTemplateDetailVo> listManagedServiceTemplates(
     @Operation(description = "view service template by id.")
-    @GetMapping(value = "/csp/service_templates/{id}", produces =
+    @GetMapping(value = "/csp/service_templates/{serviceTemplateId}", produces =
             MediaType.APPLICATION_JSON_VALUE)
     @ResponseStatus(HttpStatus.OK)
     @AuditApiRequest(methodName = "getCspFromServiceTemplateId")
     public ServiceTemplateDetailVo getRegistrationDetails(
-            @Parameter(name = "id", description = "id of service template")
-            @PathVariable("id") String id) {
-        ServiceTemplateEntity templateEntity =
-                serviceTemplateManage.getServiceTemplateDetails(UUID.fromString(id), false, true);
+            @Parameter(name = "serviceTemplateId", description = "id of service template")
+            @PathVariable("serviceTemplateId") String serviceTemplateId) {
+        ServiceTemplateEntity templateEntity = serviceTemplateManage.getServiceTemplateDetails(
+                UUID.fromString(serviceTemplateId), false, true);
         return convertToServiceTemplateDetailVo(templateEntity);
     }
 
+
+    /**
+     * List pending service template request
+     *
+     * @return service templates
+     */
+    @Tag(name = "CloudServiceProvider",
+            description = "APIs for cloud service provider to manage service templates.")
+    @Operation(description = "List service template requests history pending to review.")
+    @GetMapping(value = "/csp/service_templates/requests/pending",
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseStatus(HttpStatus.OK)
+    @AuditApiRequest(enabled = false)
+    public List<ServiceTemplateRequestVo> listPendingServiceReviewRequests(
+            @Parameter(name = "serviceTemplateId", description = "id of service template")
+            @RequestParam(name = "serviceTemplateId", required = false)
+            UUID serviceTemplateId) {
+        Csp csp = userServiceHelper.getCurrentUserManageCsp();
+        ServiceTemplateHistoryQueryModel queryModel = ServiceTemplateHistoryQueryModel.builder()
+                .csp(csp).serviceTemplateId(serviceTemplateId)
+                .status(ServiceTemplateChangeStatus.IN_REVIEW).build();
+        return serviceTemplateManage.listPendingServiceTemplateRequests(queryModel);
+    }
+
     /**
      * Review service template registration.
      *
-     * @param id                        id of service template.
      * @param reviewRegistrationRequest review request for service template registration.
      */
     @Tag(name = "CloudServiceProvider",
             description = "APIs for cloud service provider to manage service templates.")
-    @Operation(description = "Review service template registration.")
-    @PutMapping(value = "/service_templates/review/{id}", produces =
+    @Operation(description = "Review change of service template.")
+    @PutMapping(value = "/service_templates/review/change", produces =
```

Comment:
the previous URI was fine. because we do not review only  the changed templates but also initial template registrations. 
Just the changeid should be fine in the request. We can get the template Id out of it.

Reply:
The template id is not mandatory but the change id for this method.  Can we rename all `change` related to the service template to 'request', such as changeId -> requestId, ServiceTemplateChangeStatus-> ServiceTemplateRequestStatus, /service_templates/{serviceTemplateId}/history -> /service_templates/{serviceTemplateId}/requests, ServiceTemplateHistoryEntity->ServiceTemplateRequestHistoryEntity, /service_templates/request/{changeId} ->/service_templates/requests/{requestId} ,etc.

Reply:
@baixinsui yes, its a good suggestion. We can do it.